bodhi/tests/server/services/test_updates.py

```diff
@@ -599,6 +601,195 @@ def test_provenpackager_request_privs(self, publish, *args):
+    def test_provenpackager_request_ignored_by_ci(self, publish, *args):
+        "Ensure provenpackagers can change the request for any update"
+        nvr = u'bodhi-2.1-1.fc17'
+        user = User(name=u'bob')
+        self.db.add(user)
+        self.db.add(User(name=u'ralph'))  # Add a non proventester
+        self.db.add(User(name=u'someuser'))  # An unrelated user with no privs
+        self.db.flush()
+        group = self.db.query(Group).filter_by(name=u'provenpackager').one()
+        user.groups.append(group)
+
+        app = TestApp(main({}, testing=u'ralph', session=self.db, **self.app_settings))
+        up_data = self.get_update(nvr)
+        up_data['csrf_token'] = app.get('/csrf').json_body['csrf_token']
+        res = app.post_json('/updates/', up_data)
+        assert 'does not have commit access to bodhi' not in res, res
+        publish.assert_called_once_with(
+            topic='update.request.testing', msg=mock.ANY)
+
+        build = self.db.query(RpmBuild).filter_by(nvr=nvr).one()
+        build.ci_status = CiStatus.ignored
+        eq_(build.update.request, UpdateRequest.testing)
+
+        # Try and submit the update to stable as a non-provenpackager
+        app = TestApp(main({}, testing=u'ralph', session=self.db, **self.app_settings))
+        post_data = dict(update=nvr, request='stable',
+                         csrf_token=app.get('/csrf').json_body['csrf_token'])
+        res = app.post_json('/updates/%s/request' % str(nvr), post_data, status=400)
+
+        # Ensure we can't push it until it passed CI
+        eq_(res.json_body['status'], 'error')
+        eq_(res.json_body['errors'][0]['description'],
+            config.get('not_yet_tested_msg'))
+
+    @mock.patch(**mock_taskotron_results)
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch('bodhi.server.notifications.publish')
+    @mock.patch.dict('bodhi.server.config.config', {'ci.required': True})
+    def test_provenpackager_request_no_ci(self, publish, *args):
+        "Ensure provenpackagers can change the request for any update"
+        nvr = u'bodhi-2.1-1.fc17'
+        user = User(name=u'bob')
+        self.db.add(user)
+        self.db.add(User(name=u'ralph'))  # Add a non proventester
+        self.db.add(User(name=u'someuser'))  # An unrelated user with no privs
+        self.db.flush()
+        group = self.db.query(Group).filter_by(name=u'provenpackager').one()
+        user.groups.append(group)
+
+        app = TestApp(main({}, testing=u'ralph', session=self.db, **self.app_settings))
+        up_data = self.get_update(nvr)
+        up_data['csrf_token'] = app.get('/csrf').json_body['csrf_token']
+        res = app.post_json('/updates/', up_data)
+        assert 'does not have commit access to bodhi' not in res, res
+        publish.assert_called_once_with(
+            topic='update.request.testing', msg=mock.ANY)
+
+        build = self.db.query(RpmBuild).filter_by(nvr=nvr).one()
+        build.ci_status = None
```

Comment:
I believe we sorted out this question (also now documented in the doc of the CiStatus object)